lib/adapters/xhr.js

```diff
@@ -18,6 +18,13 @@ module.exports = function xhrAdapter(config) {
       delete requestHeaders['Content-Type']; // Let the browser set it
     }
 
+    if (
+      (utils.isBlob(requestData) || utils.isFile(requestData)) &&
+      requestData.type
+    ) {
+      delete requestHeaders['Content-Type']; // Let the browser set it
+    }
```

Comment:
This breaks allowing the developer to explicitly set the `Content-Type` header. For example, if I need to send `Content-Type` of `application/octet-stream` for a PNG image, the browser overrides this and instead sends `image/png`. 

For a use-case, this breaks sending attachments to the Microsoft Azure Rest API, which requires a `Content-Type` of `application/octet-stream`

Reply:
Hi @adamdehaven, thanks for the report.

Although I think this is a breaking change and not sure why this was landed in a minor version, in your case, I think actually you need to build a blob with type `application/octet-stream`, which could be done by something like `blob = blob.slice(0, blob.size, newType)`

Here is a codepen: https://codepen.io/Gerhut/pen/YzqaOQY?editors=0012

Reply:
I'm not sure why I'd need to build a blob instead when this worked fine in v0.19.x? 

Why delete the Content-Type value set in the code by the dev? If it's being explicitly set, that value should be respected. Axios should not "decide" which one to use.

Reply:
“Theoretically” speaking, the dev also "set" a content-type in the body if they put a blob in body, which goes after headers and should overwrite the former definition.

Reply:
Hey @Gerhut, when will https://github.com/axios/axios/pull/3289 be merged? The solution you mentioned in https://github.com/axios/axios/pull/1773#discussion_r486469250 is not working for IE11.